### ISSUE TITLE
how to remove version from status.storedVersions while Manually upgrading the existing objects to a new stored version

### ISSUE THREAD

**Issue by damodhar22** [kubernetes/website#18248]
**This is a Bug Report**

**Problem:**
How to remove old version from the CustomResourceDefinition status.storedVersions field? It is not clear from the document how to do this.
**Proposed Solution:**

**Page to Update:**
https://kubernetes.io/...

**Comment by damodhar22** [comment]
> /sig api-machinery

How to do it using kubectl. 

I am trying to remove the older version from list of versions of crds it's giving me following error

`The CustomResourceDefinition "crontabs.stable.example.com" is invalid: status.storedVersions[0]: Invalid value: "v3": must appear in spec.versions`

**Comment by lucky125111** [comment]
**TL;DR**
What worked for me was using golang script from [script](https://github.com/elastic/cloud-on-k8s/issues/2196#issuecomment-560318404). 

This [thread](https://github.com/elastic/cloud-on-k8s/issues/2196) describes more about removing stored versions.

**My scenario**
Creating dummy resource
1. First created the resource crd1.yaml
```yaml
  versions:
    - name: v1
      served: false
      storage: false
```

`kubectl apply -f crd1.yaml`

2. Update resource with definition of version 2 crd12.yaml
```yaml
  versions:
    - name: v1
      served: false
      storage: false
    ...
    - name: v2
      served: true
      storage: true
```

`kubectl apply -f crd12.yaml`

3. Checing stored versions
`kubectl get CustomResourceDefinition $resource_name -o yaml`

```yaml
  versions:
  - name: v1
    served: false
    storage: false
  - name: v2
    served: true
    storage: true
...
  storedVersions:
  - v1
  - v2
```

**Removing first stored version:**
1. `kubectl` - I tried to remove the first version by applying new definition crd2.yaml

```yaml
    - name: v2
      served: true
      storage: true
```

`kubectl apply -f crd2.yaml`

error:
* spec.version: Invalid value: "v1": must match the first version in spec.versions
* status.storedVersions[0]: Invalid value: "v1": must appear in spec.versions

2. Update spec version via Rest API

```bash
curl -d '[{ "op": "replace", "path":"/status/storedVersions", "value": ["v2"] }]' -H "Content-Type: application/json-patch+json" -X PATCH http://localhost:8080/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/$resource_name
```
I get as a response 200 OK, but the version hasn't been removed

3. Golang client
I got the storedVersion removed using this [script](https://github.com/elastic/cloud-on-k8s/issues/2196#issuecomment-560318404)

**Comment by sttts** [comment]
I reproduced the `kubectl edit` steps. This looks like a bug somewhere. A merge patch setting `storedVersions` to a new slice is supposed to work.

**Comment by vareti** [comment]
There is an open issue with updating status subresource using `kubectl patch`.

https://github.com/kubernetes/kubernetes/issues/67455

It might be the same issue we are seeing here with `kubectl edit`

**Comment by PG2000** [comment]
For me something like this worked:

```
kubectl patch customresourcedefinitions somecrd --subresource='status' --type='merge' -p '{"status":{"storedVersions": ["v1alpha2"]}}'
```

**Comment by booleanbetrayal** [comment]
> For me something like this worked:
> 
> ```
> kubectl patch customresourcedefinitions somecrd --subresource='status' --type='merge' -p '{"status":{"storedVersions": ["v1alpha2"]}}'
> ```

This `patch` workaround was a life-saver for us after a botched Karpenter CRD upgrade. Allowed us to clean up the resource for subsequent deletion.
